newIDE/app/src/ObjectEditor/Editors/SpriteEditor/PointsEditor/PointRow.js

```diff
@@ -6,96 +6,144 @@ import IconButton from '../../../../UI/IconButton';
-              value={props.pointY}
-              type="number"
-              id="point-y"
-              onChange={(e, value) =>
-                props.onChangePointY(parseFloat(value || 0))
+              defaultValue={props.pointName || 'Unnamed point'}
+              id={props.pointName}
+              fullWidth
+              errorText={
+                props.nameError ? 'This name is already taken' : undefined
               }
+              disabled={!props.onBlur}
+              onBlur={props.onBlur}
             />
-          ) : (
-            <Text noMargin>
-              <Trans>(auto)</Trans>
-            </Text>
-          )}
-        </TableRowColumn>
-        <TableRowColumn style={styles.toolColumn}>
-          {!!props.onRemove && (
-            <IconButton size="small" onClick={props.onRemove}>
-              <Delete />
-            </IconButton>
-          )}
-          {!!props.onEdit && (
-            <IconButton size="small" onClick={props.onEdit}>
-              <Edit />
-            </IconButton>
-          )}
-        </TableRowColumn>
-      </TableRow>
-    )}
-  </ThemeConsumer>
-);
+          </TableRowColumn>
+          <TableRowColumn style={styles.coordinateColumn}>
+            {!props.isAutomatic ? (
+              <SemiControlledTextField
+                margin="none"
+                inputStyle={
+                  props.selected
+                    ? { color: muiTheme.listItem.selectedTextColor }
+                    : undefined
+                }
+                value={roundTo(pointX, POINT_COORDINATE_PRECISION).toString()}
+                type="number"
+                id="point-x"
+                onChange={value => {
+                  const valueAsNumber = parseFloat(value);
+                  if (!isNaN(valueAsNumber))
+                    props.onChangePointX(valueAsNumber);
+                }}
+                onBlur={event => {
+                  props.onChangePointX(
+                    parseFloat(event.currentTarget.value) || 0
+                  );
+                }}
+              />
+            ) : (
+              <Text noMargin>
+                <Trans>(auto)</Trans>
+              </Text>
+            )}
+          </TableRowColumn>
+          <TableRowColumn style={styles.coordinateColumn}>
+            {!props.isAutomatic ? (
+              <SemiControlledTextField
+                margin="none"
+                inputStyle={
+                  props.selected
+                    ? { color: muiTheme.listItem.selectedTextColor }
+                    : undefined
+                }
+                value={roundTo(pointY, POINT_COORDINATE_PRECISION).toString()}
+                type="number"
+                id="point-y"
+                onChange={value => {
+                  const valueAsNumber = parseFloat(value);
+                  if (!isNaN(valueAsNumber))
+                    props.onChangePointY(valueAsNumber);
+                }}
+                onBlur={event => {
```

Comment:
Without the onBlur, a user can leave the input empty, having then a quantic state: will it be 0? will it be something else? Well, it will be the last valid value, if you were erasing `5,643`, you will have 5 as a value. Adding this onBlur, I set empty values to 0 so that it's clear to the user what it will be